wallet-new/src/Cardano/Wallet/WalletLayer/Kernel.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Wallet.WalletLayer.Kernel
+    ( bracketPassiveWallet
+    , bracketActiveWallet
+    ) where
+
+import           Universum
+
+import           Cardano.Wallet.WalletLayer.Types (ActiveWalletLayer (..), PassiveWalletLayer (..))
+
+import qualified Cardano.Wallet.Kernel as Kernel
+import           Cardano.Wallet.Kernel.Diffusion (WalletDiffusion (..))
+
+-- | Initialize the passive wallet.
+-- The passive wallet cannot send new transactions.
+bracketPassiveWallet
+    :: forall m n a. (MonadMask m, Monad n)
+    => (PassiveWalletLayer n -> m a) -> m a
+bracketPassiveWallet =
+    bracket
+        (Kernel.bracketPassiveWallet logFunction passiveWalletLayer)
+        (\_ -> return ())
+  where
+    -- | TODO(ks): Currently not implemented!
+    passiveWalletLayer _wallet =
+        pure $ PassiveWalletLayer
+            { pwlGetWalletAddresses  = error "Not implemented!"
+            , pwlGetWalletMeta       = error "Not implemented!"
+            }
+
+    -- | Empty log function.
+    logFunction _ = pure $ pure ()
```

Comment:
Let's make this an argument again (where we this bracket function we construct it anyway).